packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/direct-edit/DiagramDirectEditInput.tsx

```diff
@@ -88,6 +94,15 @@ export const DiagramDirectEditInput = ({ labelId, editingKey, onClose, transform
 
   const [renameElement, { data: editLabelData, error: editLabelError }] = useMutation(editLabelMutationOp);
 
+  useEffect(() => {
+    return () => {
+      let elementToFocus: HTMLDivElement | null = (parentElementRef.current?.parentNode as HTMLDivElement) ?? null;
```

Comment:
Why do you try to focus on parentNode while `parentElementRef.current` seems to reference the parent?